COMPATIBILITY.md

```diff
@@ -0,0 +1,80 @@
+# Compatibility Guarantee
+## Introduction
+
+The release of InfluxDB version 1 is a major milestone in the
+development of the software lifecycle. InfluxDB 1 is a stable platform
+for the growth of programs and projects that utilize InfluxDB as a
+database.
+
+## Expectations
+
+Although we expect that the vast majority of users will maintain this
+compatibility over time, it is impossible to guarantee that no future
+change will break any program. This document is an attempt to set
+expectations for the compatibility of InfluxDB 1 in the future. There
+are a number of ways in which a client that works today may fail to do
+so after a future point release. They are unlikely but worth recording.
+
+* Security. A security issue in the specification or implementation may
+  come to light whose resolution requires breaking compatibility. We
+  reserve the right to address such security issues.
+* Unspecified behavior. The InfluxDB specification tries to be explicit
+  about most properties of the database, but there are some aspects that
+  are undefined. Clients that depend on such unspecified behavior may
+  break in future releases.
+* Specification errors. If it becomes necessary to address an
+  inconsistency or incompleteness in the specification, resolving the
+  issue could affect existing clients. We reserve the right to address
+  such issues. Except for security issues, no incompatible changes to
+  the specification would be made.
+* Bugs. If the software has a bug that violates the specification, a
+  program that depends on the buggy behavior may break if the bug is
+  fixed. We reserve the right to fix such bugs.
+* Go packages outside of the `client` packages. As part of normal
+  development, exposed functions may become obsolete or may need to be
+  changed. If you need to use one of these packages, please use some
+  kind of vendoring or version pinning. Any `client` packages will
+  follow the Go 1 compatibility rules between releases.
+* Bare identifiers. InfluxQL identifiers do not need to be surrounded by
```

Comment:
It just seemed like this section and the "private identifiers" section below it cover InfluxQL specific guarantees when the other sections are higher-level exceptions. Maybe we could have separate sections for exceptions in the line protocol, API, and InfluxQL.